File: bdp-govern/metadata/atlas/dashboardv3/public/js/utils/Enums.js
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

define(["require", "backbone"], function(require) {
    'use strict';

    var Enums = {};

    Enums.auditAction = {
        ENTITY_CREATE: "Entity Created",
        ENTITY_UPDATE: "Entity Updated",
        ENTITY_DELETE: "Entity Deleted",
        CLASSIFICATION_ADD: "Classification Added",
        CLASSIFICATION_DELETE: "Classification Deleted",
        CLASSIFICATION_UPDATE: "Classification Updated",
        PROPAGATED_CLASSIFICATION_ADD: "Propagated Classification Added",
        PROPAGATED_CLASSIFICATION_DELETE: "Propagated Classification Deleted",
        PROPAGATED_CLASSIFICATION_UPDATE: "Propagated Classification Updated",
        ENTITY_IMPORT_CREATE: "Entity Created by import",
        ENTITY_IMPORT_UPDATE: "Entity Updated by import",
        ENTITY_IMPORT_DELETE: "Entity Deleted by import",
        TERM_ADD: "Term Added",
        TERM_DELETE: "Term Deleted",
        LABEL_ADD: "Label(s) Added",
        LABEL_DELETE: "Label(s) Deleted",
        ENTITY_PURGE: "Entity Purged",
        BUSINESS_ATTRIBUTE_ADD: "Business Attribute(s) Added",
        BUSINESS_ATTRIBUTE_UPDATE: "Business Attribute(s) Updated",
        BUSINESS_ATTRIBUTE_DELETE: "Business Attribute(s) Deleted",
        CUSTOM_ATTRIBUTE_UPDATE: "User-defined Attribute(s) Updated",
        TYPE_DEF_UPDATE: "Type Updated",
        TYPE_DEF_CREATE: "Type Created",
        TYPE_DEF_DELETE: "Type Deleted",
        IMPORT: "Import",
        EXPORT: "Export"
    }
    Enums.serverAudits = {
        SERVER_START: "Server Start",
        SERVER_STOP: "Server End",
        SERVER_STATE_ACTIVE: "Server State Active",
        SERVER_STATE_PASSIVE: "Server State Passive"
    }

    Enums.category = {
        PRIMITIVE: "Primitive",
        OBJECT_ID_TYPE: "Object Id type",
        ENUM: "Enum",
        STRUCT: "Struct",
        CLASSIFICATION: "Classification",
        ENTITY: "Entity",
        ARRAY: "Array",
        MAP: "Map",
        RELATIONSHIP: "Relationship",
        BUSINESS_METADATA: "Business Metadata",
        PURGE: "Purge Entities",
        IMPORT: "Import Entities",
        EXPORT: "Export Entities"
    }

    Enums.entityStateReadOnly = {
        ACTIVE: false,
        DELETED: true,
        STATUS_ACTIVE: false,
        STATUS_DELETED: true
    }

    Enums.isEntityPurged = {
        PURGED: true
    }

    Enums.lineageUrlType = {
        INPUT: 'inputs',
        OUTPUT: 'outputs',
        SCHEMA: 'schema'
    }

    Enums.searchUrlType = {
        DSL: 'dsl',
        FULLTEXT: 'basic'
    }

    Enums.profileTabType = {
        'count-frequency': "Count Frequency Distribution",
        'decile-frequency': "Decile Frequency Distribution",
        'annual': "Annual Distribution"
    }
    Enums.extractFromUrlForSearch = {
        "searchParameters": {
            "pageLimit": "limit",
            "type": "typeName",
            "tag": "classification",
            "query": "query",
            "pageOffset": "offset",
            "includeDE": "excludeDeletedEntities",
            "excludeST": "includeSubTypes",
            "excludeSC": "includeSubClassifications",
            "tagFilters": "tagFilters",
            "entityFilters": "entityFilters",
            "attributes": "attributes",
            "term": "termName"
        },
        "uiParameters": "uiParameters"
    }
    Enums.regex = {
        RANGE_CHECK: {
            "byte": {
                min: -128,
                max: 127
            },
            "short": {
                min: -32768,
                max: 32767
            },
            "int": {
                min: -2147483648,
                max: 2147483647
            },
            "long": {
                min: -9223372036854775808,
                max: 9223372036854775807
            },
            "float": {
                min: -3.4028235E38,
                max: 3.4028235E38
            },
            "double": {
                min: -1.7976931348623157E308,
                max: 1.7976931348623157E308
            }
        }
    }
    Enums.graphIcon = {
        // hive_db: { icon: "fa-database", textContent: '\uf1c0' },
        // hive_column: { icon: "fa-columns", textContent: '\uf0db' },
        // hive_table: { icon: "fa-table", textContent: '\uf0ce' }
    }
    var getTermRelationAttributes = function() {
        return {
            description: null,
            expression: null,
            steward: null,
            source: null
        }
    }
    Enums.termRelationAttributeList = {
        seeAlso: getTermRelationAttributes(),
        synonyms: getTermRelationAttributes(),
        antonyms: getTermRelationAttributes(),
        preferredTerms: getTermRelationAttributes(),
        preferredToTerms: getTermRelationAttributes(),
        replacementTerms: getTermRelationAttributes(),
        replacedBy: getTermRelationAttributes(),
        translationTerms: getTermRelationAttributes(),
        translatedTerms: getTermRelationAttributes(),
        isA: getTermRelationAttributes(),
        classifies: getTermRelationAttributes(),
        validValues: getTermRelationAttributes(),
        validValuesFor: getTermRelationAttributes()
    }
    Enums.addOnClassification = ["_ALL_CLASSIFICATION_TYPES", "_CLASSIFIED", "_NOT_CLASSIFIED"];
    Enums.addOnEntities = ["_ALL_ENTITY_TYPES"];

    Enums.stats = {
        generalData: {
            "collectionTime": "day"
        },
        Server: {
            "startTimeStamp": "day",
            "activeTimeStamp": "day",
            "upTime": "none"
        },
        ConnectionStatus: {
            "statusBackendStore": "status-html",
            "statusIndexStore": "status-html"
        },
        Notification: {
            "currentDay": "number",
            "currentDayAvgTime": "number",
            "currentDayEntityCreates": "number",
            "currentDayEntityDeletes": "number",
            "currentDayEntityUpdates": "number",
            "currentDayFailed": "number",
            "currentDayStartTime": "day",
            "currentHour": "number",
            "currentHourAvgTime": "millisecond",
            "currentHourEntityCreates": "number",
            "currentHourEntityDeletes": "number",
            "currentHourEntityUpdates": "number",
            "currentHourFailed": "number",
            "currentHourStartTime": "day",
            "lastMessageProcessedTime": "day",
            "offsetCurrent": "number",
            "offsetStart": "number",
            "previousDay": "number",
            "previousDayAvgTime": "millisecond",
            "previousDayEntityCreates": "number",
            "previousDayEntityDeletes": "number",
            "previousDayEntityUpdates": "number",
            "previousDayFailed": "number",
            "previousHour": "number",
            "previousHourAvgTime": "millisecond",
            "previousHourEntityCreates": "number",
            "previousHourEntityDeletes": "number",
            "previousHourEntityUpdates": "number",
            "previousHourFailed": "number",
            "total": "number",
            "totalAvgTime": "millisecond",
            "totalCreates": "number",
            "totalDeletes": "number",
            "totalFailed": "number",
            "totalUpdates": "number",
            "processedMessageCount": "number",
            "failedMessageCount": "number"
        }
    };
    Enums.systemAttributes = {
        "__classificationNames": "Classification(s)",
        "__createdBy": "Created By User",
        "__customAttributes": "User-defined Properties",
        "__guid": "Guid",
        "__isIncomplete": "IsIncomplete",
        "__labels": "Label(s)",
        "__modificationTimestamp": "Last Modified Timestamp",
        "__modifiedBy": "Last Modified User",
        "__propagatedClassificationNames": "Propagated Classification(s)",
        "__state": "Status",
        "__entityStatus": "Entity Status",
        "__timestamp": "Created Timestamp",
        "__typeName": "Type Name"
    };
    Enums.__isIncomplete = {
        0: "false",
        1: "true"
    };

    Enums.queryBuilderUIOperatorToAPI = {
        "=": "eq",
        "!=": "neq",
        "<": "lt",
        "<=": "lte",
        ">": "gt",
        ">=": "gte",
        "begins_with": "startsWith",
        "ends_with": "endsWith",
        "not_null": "notNull",
        "is_null": "isNull",
        "TIME_RANGE": "timerange"
    };

    Enums.queryBuilderApiOperatorToUI = _.invert(Enums.queryBuilderUIOperatorToAPI);

    Enums.queryBuilderDateRangeUIValueToAPI = {
        "Today": "TODAY",
        "Yesterday": "YESTERDAY",
        "Last 7 Days": "LAST_7_DAYS",
        "Last 30 Days": "LAST_30_DAYS",
        "This Month": "THIS_MONTH",
        "Last Month": "LAST_MONTH",
        "This Quarter": "THIS_QUARTER",
        "Last Quarter": "LAST_QUARTER",
        "This Year": "THIS_YEAR",
        "Last Year": "LAST_YEAR",
        "Last 3 Months": "LAST_3_MONTHS",
        "Last 6 Months": "LAST_6_MONTHS",
        "Last 12 Months": "LAST_12_MONTHS"
    };

    Enums.queryBuilderDateRangeAPIValueToUI = _.invert(Enums.queryBuilderDateRangeUIValueToAPI);

    return Enums;
});

File: govern/meta-data/atlas-lineage/atlas-3.0.0-SNAPSHOT/intg/src/main/python/apache_atlas/model/enums.py
#!/usr/bin/env/python
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicabwle law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
import enum


class AtlasTermAssignmentStatus(enum.Enum):
    DISCOVERED = 0
    PROPOSED = 1
    IMPORTED = 2
    VALIDATED = 3
    DEPRECATED = 4
    OBSOLETE = 5
    OTHER = 6


class AtlasTermRelationshipStatus(enum.Enum):
    DRAFT = 0
    ACTIVE = 1
    DEPRECATED = 2
    OBSOLETE = 3
    OTHER = 99


class TypeCategory(enum.Enum):
    PRIMITIVE = 0
    OBJECT_ID_TYPE = 1
    ENUM = 2
    STRUCT = 3
    CLASSIFICATION = 4
    ENTITY = 5
    ARRAY = 6
    MAP = 7
    RELATIONSHIP = 8
    BUSINESS_METADATA = 9


class Cardinality(enum.Enum):
    SINGLE = 0
    LIST = 1
    SET = 2


class Condition(enum.Enum):
    AND = 0
    OR = 1


class EntityOperation(enum.Enum):
    CREATE = 0
    UPDATE = 1
    PARTIAL_UPDATE = 2
    DELETE = 3
    PURGE = 4


class EntityStatus(enum.Enum):
    ACTIVE = 0
    DELETED = 1
    PURGED = 2


class IndexType(enum.Enum):
    DEFAULT = 0
    STRING = 1


class LineageDirection(enum.Enum):
    INPUT = 0
    OUTPUT = 1
    BOTH = 2


class Operator(enum.Enum):
    LT = ("<", "lt")
    GT = ('>', 'gt')
    LTE = ('<=', 'lte')
    GTE = ('>=', 'gte')
    EQ = ('=', 'eq')
    NEQ = ('!=', 'neq')
    IN = ('in', 'IN')
    LIKE = ('like', 'LIKE')
    STARTS_WITH = ('startsWith', 'STARTSWITH', 'begins_with', 'BEGINS_WITH')
    ENDS_WITH = ('endsWith', 'ENDSWITH', 'ends_with', 'ENDS_WITH')
    CONTAINS = ('contains', 'CONTAINS')
    NOT_CONTAINS = ('not_contains', 'NOT_CONTAINS')
    CONTAINS_ANY = ('containsAny', 'CONTAINSANY', 'contains_any', 'CONTAINS_ANY')
    CONTAINS_ALL = ('containsAll', 'CONTAINSALL', 'contains_all', 'CONTAINS_ALL')
    IS_NULL = ('isNull', 'ISNULL', 'is_null', 'IS_NULL')
    NOT_NULL = ('notNull', 'NOTNULL', 'not_null', 'NOT_NULL')


class PropagateTags(enum.Enum):
    NONE = 0
    ONE_TO_TWO = 1
    TWO_TO_ONE = 2
    BOTH = 3


class QueryType(enum.Enum):
    DSL = 0
    FULL_TEXT = 1
    GREMLIN = 2
    BASIC = 3
    ATTRIBUTE = 4
    RELATIONSHIP = 5


class RelationshipCategory(enum.Enum):
    ASSOCIATION = 0
    AGGREGATION = 1
    COMPOSITION = 2


class RelationshipStatus(enum.Enum):
    ACTIVE = 0
    DELETED = 1


class SavedSearchType(enum.Enum):
    BASIC = 0
    ADVANCED = 1


class SortOrder(enum.Enum):
    ASCENDING = 0
    DESCENDING = 1


class SortType(enum.Enum):
    NONE = 0
    ASC = 1
    DESC = 2
